TKeazirian.HTTPServer/Router/Router.cs

```diff
@@ -0,0 +1,79 @@
+namespace TKeazirian.HTTPServer.Router;
+
+using Handler;
+using Response;
+using Request;
+
+public class Router
+{
+    private readonly RoutesConfig _routesConfig;
+
+    public Router(RoutesConfig routesConfig)
+    {
+        _routesConfig = routesConfig;
+    }
+
+    public Response Route(Request request)
+    {
+        if (RouteConfigContainsPath(request) && !IsHttpMethodAllowed(request))
+        {
+            return new NotImplementedResponse().BuildNotImplementedResponse();
+        }
+
+        if (RouteConfigContainsPath(request) && IsHttpMethodAllowed(request))
+        {
+            var handler = GetHandler(request);
+
+            if (IsHeadRequest(request) && HandlerContainsGetMethod(handler))
+            {
+                return new HeadResponse().BuildHeadResponse(handler, request);
+            }
+
+            if (IsOptionsResponse(request) && HandlerContainsGetMethod(handler))
+            {
+                return new OptionsResponse().BuildOptionsResponse(request, _routesConfig);
+            }
+
+            return handler.HandleResponse(request);
+        }
+
+        return new ResourceNotFoundHandler().HandleResponse(request);
+    }
+
+    private Handler GetHandler(Request request)
+    {
+        return _routesConfig.Routes[request.GetRequestPath()];
+    }
+
+    private bool RouteConfigContainsPath(Request request)
+    {
+        return _routesConfig.Routes.ContainsKey(request.GetRequestPath());
+    }
+
+    private static bool HandlerContainsGetMethod(Handler handler)
+    {
+        return handler.AllowedHttpMethods().Contains("GET");
+    }
+
+    private static bool IsOptionsResponse(Request request)
+    {
+        return request.GetRequestMethod() == "OPTIONS";
+    }
+
+    private static bool IsHeadRequest(Request request)
+    {
+        return request.GetRequestMethod() == "HEAD";
+    }
```

Comment:
when the name of the method and the one line command that is contained within are basically the same; ask yourself if the method is even required in the first place.

Is this just a lot of uneeded abstraction?

Reply:
a good example is:
`HandlerContainsGetMethod(Handler handler)`
`handler.AllowedHttpMethods().Contains("GET")`

They look almost identical, just slight changes in the order of the words.  In my opinion, this shows that you chose really good names for the methods and how they function.  Why not use them as is?

Reply:
Got it - makes sense!!